Makefile.cache

```diff
@@ -0,0 +1,637 @@
+#             the build framework is part of sonic main repo.
+#             The build framework downloads the zipped source content from the web,
+#             applies a series of patches (if applicable) on the downloaded source code,
+#             compiles the source and generates the final .deb package(s).
+#                 Eg: bash, ntp, etc
+#
+#         2. Module source code is maintained as a submodule(SM) in the sonic repo.
+#             Eg: Frr, swss, Linux etc
+#
+#
+# The sonic build framework uses the module .deb packages that are generated as part of build and
+# creates  the set of target docker images and the final binary image for distribution.
+#
+# The caching framework provides the method to cache the module .deb packages and docker images
+# into a cache location by tracking module dependency information.
+#
+# A module can have a set of dependency files, for example Makefiles, source files, scripts, dpkg control files, etc.
+# The caching is done based on the SHA hash value of the module dependency file contents. If one of the
+# dependency files is changed, the corresponding cache file is also changed. So it automatically creates the new cache file
+# for that module.
+#
+# It provides two levels of caching.
+#     Global cache := Module doesnt have any local changes.
+#     Local cache  := Used for target when one of its dependency file is modified.
+#
+#
+# Steps for adding new module is given as a template.
+# Template File : rules/template.dep
+#
+
+
+# Common files and FLAGS
+SONIC_COMMON_FILES_LIST :=  $(if $(wildcard nocommon),, .platform slave.mk rules/functions Makefile.cache)
+SONIC_COMMON_FLAGS_LIST :=  $(CONFIGURED_PLATFORM) $(SONIC_DPKG_CACHE_METHOD) \
+                            $(SONIC_DPKG_CACHE_SOURCE) $(SONIC_DEBUGGING_ON) \
+                            $(SONIC_PROFILING_ON) $(SONIC_SANITIZER_ON)
+SONIC_COMMON_DPKG_LIST  :=  debian/control debian/changelog debian/rules \
+                            debian/compat debian/install debian/copyright
+SONIC_COMMON_BASE_FILES_LIST  := sonic-slave-jessie/Dockerfile.j2 sonic-slave-jessie/Dockerfile.user \
+                                 sonic-slave-stretch/Dockerfile.j2 sonic-slave-stretch/Dockerfile.user
+
+
+
+include $(RULES_PATH)/*.dep
+
+
+ifneq ($(CONFIGURED_PLATFORM), undefined)
+ifeq ($(PDDF_SUPPORT), y)
+include $(PLATFORM_PDDF_PATH)/rules.dep
+endif
+
+-include $(PLATFORM_PATH)/rules.dep
+endif
+
+$(info "SONIC_DPKG_CACHE_METHOD"         : "$(SONIC_DPKG_CACHE_METHOD)")
+ifeq ($(SONIC_DPKG_CACHE_METHOD),cache)
+$(info "DPKG_CACHE_PATH"                 : "$(SONIC_DPKG_CACHE_SOURCE)")
+endif
+$(info )
+
+
+###############################################################################
+## Canned sequences
+###############################################################################
+
+SONIC_DPKG_CACHE_DIR    := /dpkg_cache
+MOD_CACHE_LOCK_SUFFIX   := cache_access
+MOD_CACHE_LOCK_TIMEOUT  := 3600
```

Comment:
>MOD_CACHE_LOCK_TIMEOUT [](start = 0, length = 22)

Is it possible to remove the lock? A jenkins work are expected to run multiple jobs and  they are sharing the same cache, if any deadlock, hard to troubleshoot and inconvenient.

Reply:
It is required when the cache location is shared across multiple users.